icu4c/source/i18n/unicode/numberformatter.h

```diff
@@ -2663,14 +2663,28 @@ class U_I18N_API LocalizedNumberFormatter
 #endif
 
 /**
- * The result of a number formatting operation. This class allows the result to be exported in several data types,
- * including a UnicodeString and a FieldPositionIterator.
+ * Represents all the grammatical class nouns that are supported by CLDR.
+ */
+enum NounClass {
+    NEUTER = 0,
+    FEMININE = 1,
+    MASCULINE = 2,
+    ANIMATE = 3,
+    INANIMATE = 4,
+    PERSONAL = 5,
+    COMMON = 6
+};
```

Comment:
We need to add an element called `EMPTY` or `NOT_SUPPORTED` in case of the language does not support it yet.

Reply:
Should we distinguish between EMPTY and NEUTER?

In Rust we could return `Option<NounClass>`.

Reply:
I will put the rust option on my mind while adding inflection to the ICU4X.